### PR TITLE
add a few tests

### PR DESCRIPTION
**Pull request by araskachoi**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/types/statedb_test.go [hunk 1]
```diff
@@ -60,3 +67,34 @@ func (suite *StateDBTestSuite) TestBloomFilter() {
...
+
+	resp := stateDB.WithContext(suite.ctx)
+	suite.Require().NotNil(suite.T(), resp)
+}
+
+func (suite *StateDBTestSuite) TestStateDBBalance() {
+	checkTx := false
+
+	suite.app = app.Setup(checkTx)
+	suite.ctx = suite.app.BaseApp.NewContext(checkTx, abci.Header{Height: 1})
```
Comment (fedekunze) [inline x/evm/types/statedb_test.go]: You can remove these lines since the `SetupTest` does it automatically

File: x/evm/types/statedb_test.go [hunk 2]
```diff
@@ -60,3 +67,34 @@ func (suite *StateDBTestSuite) TestBloomFilter() {
 	suite.Require().True(ethtypes.BloomLookup(bloomFilter, contractAddress))
 	suite.Require().False(ethtypes.BloomLookup(bloomFilter, ethcmn.BigToAddress(big.NewInt(2))))
 }
+
+func (suite *StateDBTestSuite) TestStateDBWithContext() {
```
Comment (fedekunze) [inline x/evm/types/statedb_test.go]: you can define the `stateDB` as a field for the testing suite (i.e on `SetupTest`) and remove this test entirely 👍 

```go
suite.stateDB = suite.app.EvmKeeper.CommitStateDB.WithContext(suite.ctx)
```

File: x/evm/types/statedb_test.go [hunk 3]
```diff
@@ -60,3 +67,34 @@ func (suite *StateDBTestSuite) TestBloomFilter() {
...
+
+	suite.app = app.Setup(checkTx)
+	suite.ctx = suite.app.BaseApp.NewContext(checkTx, abci.Header{Height: 1})
+
+	stateDB := suite.app.EvmKeeper.CommitStateDB
+
+	priv, err := crypto.GenerateKey()
+	if err != nil {
+		panic(err)
+	}
```
Comment (fedekunze) [inline x/evm/types/statedb_test.go]:
```suggestion
	suite.Require().NoError(err)
```

**Review by fedekunze — COMMENTED** [review]
Look good. See my comment. I also recommend you to write table tests since it's easier to cover all the cases/lines. You can see an example on `x/evm/types/msg_test.go`

File: x/evm/types/statedb_test.go (hunk 3 above)
Comment (araskachoi) [reply x/evm/types/statedb_test.go]: fixed!

File: x/evm/types/statedb_test.go (hunk 2 above)
Comment (araskachoi) [reply x/evm/types/statedb_test.go]: i believe i did this correctly, a quick check to the changes would be appreciated

File: x/evm/types/statedb_test.go (hunk 1 above)
Comment (araskachoi) [reply x/evm/types/statedb_test.go]: fixed

**Review by noot — COMMENTED** [review]
looks good :D

**Comment by fedekunze** [comment]
@araskachoi can you fix the failing tests to merge this as is? We can continue with the other tests afterwards

**Pull request merged**